### ISSUE TITLE
Window stops responding when switching to accessible buffer in a terminal with a lot of history

### ISSUE THREAD

**Issue by roblourens** [microsoft/vscode#181081]
- Set `"terminal.integrated.scrollback": 150000`- I have this, not sure if it's required
- Run a command that produces a ton of output- in my case it was a ripgrep search, but it probably doesn't matter
- shift+tab to the accessible buffer
- The renderer process is stuck at 100% and the window does not respond. The terminal ends up showing the accessible buffer at some point but the window is not responding to clicks or scrolls anywhere

Trying to get a profile but since the window is totally blocked, it can't complete

**Comment by Tyriar** [comment]
> Set "terminal.integrated.scrollback": 150000- I have this, not sure if it's required

@roblourens the scrollback gets pre-allocated so you are wasting a lot of memory by doing this. I'd suggest using a more sane number like 10000 where this should have a much smaller impact.

**Comment by roblourens** [comment]
Ok. Maybe that's the same as what was behind my other terminal perf issue then.

**Comment by roblourens** [comment]
[renderer-af4b45.cpuprofile.zip](https://github.com/microsoft/vscode/files/11777537/renderer-af4b45.cpuprofile.zip)

Here's a trace where the window hung for a few seconds and then the automatic renderer profiling kicked in. I took a 0 off the scrollback so it's only `15000` here, I'll set it smaller now though

**Comment by Tyriar** [comment]
I can't seem to load the cpuprofile, it just comes up empty in Chrome and Edge

**Comment by roblourens** [comment]
I think the "javascript profiler" tool is deprecated and it doesn't even show up for me in Edge anymore. I was able to load it in vscode's devtools so maybe try that. This is one of the autogenerated profiles that are created through the debug API and maybe we can't get traces for the "Performance" tab.

<img width="336" alt="image" src="https://github.com/microsoft/vscode/assets/323878/67a6ab2b-25ee-4e36-b816-bee6f8e3ba73">

**Comment by Tyriar** [comment]
Seems monaco related, not xterm.js

**Comment by roblourens** [comment]
Oh, that's where the stack is? I don't know what file I had open, maybe it was one with very long lines. You can close this if it's not actionable

**Comment by meganrogge** [comment]
Closing as no users have reported this
